src/service/resources-impl.js

```diff
@@ -407,6 +410,12 @@ export class Resources {
    * @private
    */
   buildReadyResources_() {
+    // Avoid cases where elements add more elements inside of them
+    // and cause an infinite loop of building - see #3354 for details.
+    if (this.isCurrentlyBuildingPendingResources_) {
+      return;
+    }
+    this.isCurrentlyBuildingPendingResources_ = true;
```

Comment:
With the try-finally we should not get stuck. Maybe I misunderstand the question.